packages/build/src/log/messages/core_commands.js

```diff
@@ -1,17 +1,72 @@
 'use strict'
 
-const { log, logArray, logError } = require('../logger')
+const path = require('path')
+
+const { log, logArray, logErrorSubHeader } = require('../logger')
 const { THEME } = require('../theme')
 
-const logExperimentalEsbuildParameter = function (logs, envVariableName) {
-  logError(
-    logs,
-    `
-You've opted in to an experimental bundling mechanism via the "${envVariableName}" environment variable.
-We recommend against using this functionality in production sites since its behavior may change or it may be removed entirely.
-For the latest updates about function bundling, please visit our Community forum: community.netlify.com.
-`,
-  )
+const logBundleErrorObject = ({ logs, object }) => {
+  const { location = {}, text } = object
+  const { column, file, line, lineText } = location
+  const locationMessage = `${file} ${line}:${column}`
+  const logMessage = `> ${lineText}
+  (in ${locationMessage})
+
+  ${text}
+  
+  `
+
+  log(logs, logMessage, { indent: true })
+}
+
+const logEsbuildBundleResult = ({ functionName, logs, result }) => {
+  const { bundlerErrors = [], bundlerWarnings = [] } = result
+
+  if (bundlerErrors.length !== 0) {
+    logErrorSubHeader(logs, `Bundling function \`${functionName}\` has failed:`)
+
+    bundlerErrors.forEach((bundlerError) => {
+      logBundleErrorObject({ logs, object: bundlerError })
+    })
+  }
+
+  if (bundlerWarnings.length !== 0) {
+    logErrorSubHeader(logs, `Bundling function \`${functionName}\` has produced some warnings:`)
+
+    bundlerWarnings.forEach((bundlerWarning) => {
+      logBundleErrorObject({ logs, object: bundlerWarning })
+    })
+  }
+}
+
+const logZISIBundleResult = ({ functionName, logs, result }) => {
+  const { bundlerErrors = [] } = result
+
+  if (bundlerErrors.length === 0) {
+    return
+  }
+
+  logErrorSubHeader(logs, `Fallback bundler used for function \`${functionName}\`:`)
+
+  bundlerErrors.forEach((bundlerError) => {
+    logBundleErrorObject({ logs, object: bundlerError })
+  })
+}
+
+const logBundleResults = ({ logs, results = [] }) => {
+  results.forEach((result) => {
+    if (result.runtime !== 'js' || !result.path) {
+      return
+    }
+
+    const functionName = path.basename(result.path)
```

Comment:
Would there be some cases where the Function name might not be the filename?
For example, when the Function is a directory. Or when the error happens inside a required file or dependency?

Reply:
`result.path` is the name of the ZIP file returned from `zip-it-and-ship-it`. So even if the function is at `foo/index.js`, the zip file should be `foo.zip`.

As for the error happening inside a dependency, it doesn't really matter because `path` will still be the path to the function zip, not the offending source file.

Does that make sense?

Reply:
Oh, I thought this was the path to the source file with the error, not the zip archive. This makes sense!